research/07-realisation.tex

```diff
@@ -1,14 +1,54 @@
 \section{Realisation of the Product}
 \label{sec:realisation-product}
 
-\subsection{Methodologies used during the project}
-\label{ssec:methodologies}
+The previous section analysed the requirements for the product as set in
+\cref{sec:requirement-analysis}. The purpose of this section is to discuss the
+frameworks and tools used during the development of the product.
+
+\subsection{Development frameworks}
+\label{ssec:frameworks}
+
+Spoofax is an already existing project. In order for the product to be as easily
+integrated as possible, most of the tooling used by Spoofax will be reused. This
+means that the Java programming language is to be used, together with Maven for
+the build environment, JUnit for unit tests and TravisCI for continuous
+integration. On top of this, Spoofax uses a few open source Java libraries as
+can be seen in the list below. It is most likely that these will be used in the
+product as well.
+
+\begin{itemize}
+  \item Guice, a lightweight dependency injection framework;
+  \item Guava, provides general utilities missing in the JDK;
+  \item RxJava, a library for composing asynchronous and event-based programs
+    using observable sequences;
+  \item Apache VFS, a library for accessing various different file systems;
+  \item EhCache, a standards-based cache that boosts performance;
+  \item Apache's logging library.
+\end{itemize}
+
+If new libraries are to be used during the development of the project, it is
+important that these are under a license compatible with the LGPL under which
+Spoofax is licensed. Compatible licenses include, but are not limited to, the
+LGPL, MIT, BSD and Apache licenses.
 
 \subsection{Development tools}
 \label{ssec:tools}
 
-\subsection{Available APIs from Spoofax}
-\label{ssec:available-apis-from}
+% TODO: explain why git (e.g. because spoofax already, or because ...)?
```

Comment:
Why git? Because it's the de-facto standard. I don't think there's any need to explain this.